### ISSUE TITLE
Transformer model breaks when scaling is set to `False`

### ISSUE THREAD

**Issue by abdulfatir** [awslabs/gluon-ts#2743]
## Description

The MXNet transformer model breaks when `scaling=False` is used. The following code works as expected with `scaling=True` (default).

## To Reproduce

```python
from gluonts.mx import TransformerEstimator, Trainer
from gluonts.dataset.repository.datasets import get_dataset

dataset = get_dataset("airpassengers")

tft_default = TransformerEstimator(
    freq=dataset.metadata.freq,
    prediction_length=dataset.metadata.prediction_length,
    trainer=Trainer(epochs=10),
    scaling=False,
)
predictor_tft_default = tft_default.train(
    dataset.train,
)

```

## Error message or code output

```
  0%|          | 0/50 [00:00<?, ?it/s]/Users/ansarnd/miniconda3/envs/gluonts/lib/python3.8/site-packages/mxnet/gluon/block.py:825: UserWarning: The 3-th input to HybridBlock is not used by any computation. Is this intended?
  out = self.forward(*args)

  0%|          | 0/50 [00:00<?, ?it/s]
infer_shape error. Arguments:
  data0: (32, 1)
  data1: (32, 49, 2)
  data2: (32, 49)
  data3: (32, 49)
  data4: (32, 12, 2)
  data5: (32, 12)
Traceback (most recent call last):
  File "/Users/ansarnd/miniconda3/envs/gluonts/lib/python3.8/site-packages/mxnet/gluon/block.py", line 1102, in _build_cache
    params[name].data()
  File "/Users/ansarnd/miniconda3/envs/gluonts/lib/python3.8/site-packages/mxnet/gluon/parameter.py", line 574, in data
    return self._check_and_get(self._data, ctx)
  File "/Users/ansarnd/miniconda3/envs/gluonts/lib/python3.8/site-packages/mxnet/gluon/parameter.py", line 230, in _check_and_get
    raise DeferredInitializationError(
mxnet.gluon.parameter.DeferredInitializationError: Parameter 'dec_inputlayer0_dense0_weight' has not been initialized yet because initialization was deferred. Actual initialization happens during the first forward pass. Please pass one batch of data through the network before accessing Parameters. You can also avoid deferred initialization by specifying in_units, num_features, etc., for network layers.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Users/ansarnd/miniconda3/envs/gluonts/lib/python3.8/site-packages/mxnet/gluon/block.py", line 1192, in _deferred_infer_shape
    self.infer_shape(*args)
  File "/Users/ansarnd/miniconda3/envs/gluonts/lib/python3.8/site-packages/mxnet/gluon/block.py", line 1410, in infer_shape
    self._infer_attrs('infer_shape', 'shape', *args)
  File "/Users/ansarnd/miniconda3/envs/gluonts/lib/python3.8/site-packages/mxnet/gluon/block.py", line 1398, in _infer_attrs
    arg_attrs, _, aux_attrs = getattr(out, infer_fn)(
  File "/Users/ansarnd/miniconda3/envs/gluonts/lib/python3.8/site-packages/mxnet/symbol/symbol.py", line 1101, in infer_shape
    res = self._infer_shape_impl(False, *args, **kwargs)
  File "/Users/ansarnd/miniconda3/envs/gluonts/lib/python3.8/site-packages/mxnet/symbol/symbol.py", line 1250, in _infer_shape_impl
    check_call(infer_func(
  File "/Users/ansarnd/miniconda3/envs/gluonts/lib/python3.8/site-packages/mxnet/base.py", line 246, in check_call
    raise get_last_ffi_error()
mxnet.base.MXNetError: Traceback (most recent call last):
  File "../src/c_api/c_api_symbolic.cc", line 539
MXNetError: InferShapeKeyword argument name data3 not found.
Candidate arguments:
	[0]data2
	[1]data5
	[2]data1
	[3]data4
	[4]data0
	[5]transformertrainingnetwork0_featureembedder0_cat_0_embedding_weight
	[6]dec_inputlayer0_dense0_weight
	[7]dec_inputlayer0_dense0_bias
	[8]dec_layernormalization0_layernorm0_gamma
	[9]dec_layernormalization0_layernorm0_beta
	[10]dec_multiheadselfattention_dense1_weight
	[11]dec_multiheadselfattention_dense1_bias
	[12]dec_multiheadselfattention_dense0_weight
	[13]dec_multiheadselfattention_dense0_bias
	[14]dec_layernormalization1_layernorm0_gamma
	[15]dec_layernormalization1_layernorm0_beta
	[16]dec_multiheadattention_dense1_weight
	[17]dec_multiheadattention_dense1_bias
	[18]enc_inputlayer0_dense0_weight
	[19]enc_inputlayer0_dense0_bias
	[20]enc_layernormalization0_layernorm0_gamma
	[21]enc_layernormalization0_layernorm0_beta
	[22]enc_multiheadselfattention_dense1_weight
	[23]enc_multiheadselfattention_dense1_bias
	[24]enc_multiheadselfattention_dense0_weight
	[25]enc_multiheadselfattention_dense0_bias
	[26]enc_layernormalization1_layernorm0_gamma
	[27]enc_layernormalization1_layernorm0_beta
	[28]enc_transformerfeedforward_dense0_weight
	[29]enc_transformerfeedforward_dense0_bias
	[30]enc_transformerfeedforward_dense1_weight
	[31]enc_transformerfeedforward_dense1_bias
	[32]enc_layernormalization2_layernorm0_gamma
	[33]enc_layernormalization2_layernorm0_beta
	[34]dec_multiheadattention_dense2_weight
	[35]dec_multiheadattention_dense2_bias
	[36]dec_multiheadattention_dense3_weight
	[37]dec_multiheadattention_dense3_bias
	[38]dec_multiheadattention_dense0_weight
	[39]dec_multiheadattention_dense0_bias
	[40]dec_layernormalization2_layernorm0_gamma
	[41]dec_layernormalization2_layernorm0_beta
	[42]dec_transformerfeedforward_dense0_weight
	[43]dec_transformerfeedforward_dense0_bias
	[44]dec_transformerfeedforward_dense1_weight
	[45]dec_transformerfeedforward_dense1_bias
	[46]dec_layernormalization3_layernorm0_gamma
	[47]dec_layernormalization3_layernorm0_beta
	[48]transformertrainingnetwork0_None_distr_nu_weight
	[49]transformertrainingnetwork0_None_distr_nu_bias
	[50]transformertrainingnetwork0_None_distr_sigma_weight
	[51]transformertrainingnetwork0_None_distr_sigma_bias
	[52]transformertrainingnetwork0_None_distr_mu_weight
	[53]transformertrainingnetwork0_None_distr_mu_bias



During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "run.py", line 12, in <module>
    predictor_tft_default = tft_default.train(
  File "/Users/ansarnd/Documents/repos/gluon-ts/src/gluonts/mx/model/estimator.py", line 237, in train
    return self.train_model(
  File "/Users/ansarnd/Documents/repos/gluon-ts/src/gluonts/mx/model/estimator.py", line 214, in train_model
    self.trainer(
  File "/Users/ansarnd/Documents/repos/gluon-ts/src/gluonts/mx/trainer/_base.py", line 420, in __call__
    epoch_loss = loop(
  File "/Users/ansarnd/Documents/repos/gluon-ts/src/gluonts/mx/trainer/_base.py", line 286, in loop
    _ = net(*batch.values())
  File "/Users/ansarnd/miniconda3/envs/gluonts/lib/python3.8/site-packages/mxnet/gluon/block.py", line 825, in __call__
    out = self.forward(*args)
  File "/Users/ansarnd/miniconda3/envs/gluonts/lib/python3.8/site-packages/mxnet/gluon/block.py", line 1492, in forward
    return self._call_cached_op(x, *args)
  File "/Users/ansarnd/miniconda3/envs/gluonts/lib/python3.8/site-packages/mxnet/gluon/block.py", line 1200, in _call_cached_op
    self._build_cache(*args)
  File "/Users/ansarnd/miniconda3/envs/gluonts/lib/python3.8/site-packages/mxnet/gluon/block.py", line 1104, in _build_cache
    self._deferred_infer_shape(*args)
  File "/Users/ansarnd/miniconda3/envs/gluonts/lib/python3.8/site-packages/mxnet/gluon/block.py", line 1196, in _deferred_infer_shape
    raise ValueError(error_msg)
ValueError: Deferred initialization failed because shape cannot be inferred. Traceback (most recent call last):
  File "../src/c_api/c_api_symbolic.cc", line 539
MXNetError: InferShapeKeyword argument name data3 not found.
Candidate arguments:
	[0]data2
	[1]data5
	[2]data1
	[3]data4
	[4]data0
	[5]transformertrainingnetwork0_featureembedder0_cat_0_embedding_weight
	[6]dec_inputlayer0_dense0_weight
	[7]dec_inputlayer0_dense0_bias
	[8]dec_layernormalization0_layernorm0_gamma
	[9]dec_layernormalization0_layernorm0_beta
	[10]dec_multiheadselfattention_dense1_weight
	[11]dec_multiheadselfattention_dense1_bias
	[12]dec_multiheadselfattention_dense0_weight
	[13]dec_multiheadselfattention_dense0_bias
	[14]dec_layernormalization1_layernorm0_gamma
	[15]dec_layernormalization1_layernorm0_beta
	[16]dec_multiheadattention_dense1_weight
	[17]dec_multiheadattention_dense1_bias
	[18]enc_inputlayer0_dense0_weight
	[19]enc_inputlayer0_dense0_bias
	[20]enc_layernormalization0_layernorm0_gamma
	[21]enc_layernormalization0_layernorm0_beta
	[22]enc_multiheadselfattention_dense1_weight
	[23]enc_multiheadselfattention_dense1_bias
	[24]enc_multiheadselfattention_dense0_weight
	[25]enc_multiheadselfattention_dense0_bias
	[26]enc_layernormalization1_layernorm0_gamma
	[27]enc_layernormalization1_layernorm0_beta
	[28]enc_transformerfeedforward_dense0_weight
	[29]enc_transformerfeedforward_dense0_bias
	[30]enc_transformerfeedforward_dense1_weight
	[31]enc_transformerfeedforward_dense1_bias
	[32]enc_layernormalization2_layernorm0_gamma
	[33]enc_layernormalization2_layernorm0_beta
	[34]dec_multiheadattention_dense2_weight
	[35]dec_multiheadattention_dense2_bias
	[36]dec_multiheadattention_dense3_weight
	[37]dec_multiheadattention_dense3_bias
	[38]dec_multiheadattention_dense0_weight
	[39]dec_multiheadattention_dense0_bias
	[40]dec_layernormalization2_layernorm0_gamma
	[41]dec_layernormalization2_layernorm0_beta
	[42]dec_transformerfeedforward_dense0_weight
	[43]dec_transformerfeedforward_dense0_bias
	[44]dec_transformerfeedforward_dense1_weight
	[45]dec_transformerfeedforward_dense1_bias
	[46]dec_layernormalization3_layernorm0_gamma
	[47]dec_layernormalization3_layernorm0_beta
	[48]transformertrainingnetwork0_None_distr_nu_weight
	[49]transformertrainingnetwork0_None_distr_nu_bias
	[50]transformertrainingnetwork0_None_distr_sigma_weight
	[51]transformertrainingnetwork0_None_distr_sigma_bias
	[52]transformertrainingnetwork0_None_distr_mu_weight
	[53]transformertrainingnetwork0_None_distr_mu_bias
```

## Environment
- Operating system: MacOS/Ubuntu
- Python version: 3.8
- GluonTS version: `dev`
- MXNet version: 1.9.1

(Add as much information about your environment as possible, e.g. dependencies versions.)

**Comment by lostella** [comment]
If you set `hybridize=False` when constructing the `Trainer` and the issue persists, you may be able to debug it

**Comment by abdulfatir** [comment]
_"may be"_

**Comment by abdulfatir** [comment]
Wow, this crazy: https://github.com/apache/mxnet/issues/14373 🤯 
Everything breaks because `NOPScaler` does not use `observed_indicator` in the graph. Give me back my 3 hours. 😭

**Comment by abdulfatir** [comment]
Fixed by #2744
